runtime/autoload/remote/define.vim

```diff
@@ -169,14 +169,40 @@ function! remote#define#FunctionOnChannel(channel, method, sync, name, opts)
   exe function_def
 endfunction
 
+let s:busy = {}
+let s:pending_notifications = {}
 
 function! s:GetRpcFunction(sync)
-  if a:sync
-    return 'rpcrequest'
+  if a:sync ==# 'urgent'
+    return 'rpcnotify'
+  elseif a:sync
+    return 'remote#define#request'
   endif
-  return 'rpcnotify'
+  return 'remote#define#notify'
 endfunction
 
+function! remote#define#notify(chan, ...)
```

Comment:
can this and `remote#define#request` be script-local? `s:notify`, `s:request`

Reply:
It is called by functions outside of this file (even though they are "generated" inside it), so even if it is possible, "sematically" it is wrong (until vimL gets hygenic macros).